src/Abstractions/TaskOrchestrationContext.cs

```diff
@@ -385,7 +385,7 @@ public virtual Task CallSubOrchestratorAsync(TaskName orchestratorName, TaskOpti
     /// </summary>
     /// <param name="categoryName">The logger's category name.</param>
     /// <returns>An instance of <see cref="ILogger"/> that is replay-safe.</returns>
-    public ILogger CreateReplaySafeLogger(string categoryName)
+    public virtual ILogger CreateReplaySafeLogger(string categoryName)
```

Comment:
This is fine, but you can instead mock `LoggerFactory` to accomplish the same goal (or supply `NullLoggerFactory`)

https://github.com/microsoft/durabletask-dotnet/blob/7eef072e11fa4d4f95bc4c198b7d24828c967525/src/Abstractions/TaskOrchestrationContext.cs#L71

Reply:
@jviau - not sure I follow. Naively speaking, I can't mock a `protected` property so I'm unsure how I'd mock this `ILoggerFactory` inside the orchestration context. I suspect you're thinking of something slightly different.

Reply:
You can via one of two ways:

1. Derive the type and set it to what you want
2. OR use `Moq.Protected` -> `taskOrchestrationContextMock.Protected().{rest of code}`

Reply:
Ah, `Moq.Protected` looks neat. I too on your advice to derive types to help me with the client sample. I'd appreacite another pass, but I'll probably keep this bit intact for now.